packages/engine/paima-funnel/src/funnels/carp/funnel.ts

```diff
@@ -151,10 +151,16 @@ export class CarpFunnel extends BaseFunnel implements ChainFunnel {
       lastTimestamp = cachedState.timestamp;
     } else {
       lastTimestamp = await timeout(
-        this.sharedData.web3.eth.getBlock(blockHeight - 1),
+        this.sharedData.mainNetworkApi.getBlock(blockHeight - 1),
         DEFAULT_FUNNEL_TIMEOUT
       );
 
+      if (!lastTimestamp) {
+        // FIXME: this funnel shouldn't actually be doing this, it should be
```

Comment:
?

Reply:
I didn't want to change this here since it's unrelated, and it probably doesn't make a huge difference anyway. Basically the parallel evm funnel and the mina funnel store a checkpoint in the db to know where to resume pulling data after a restart (we need to get everything with a timestamp > than the latest synchronized block).

The carp funnel instead is instead getting (again) the previous block through the base funnel's rpc. It's not much of a deal because it's just done once, but I prefer the other option.